### DISCUSSION TITLE
Is there a way to set X_VCPKG_RPATH_KEEP_SYSTEM_PATHS to ON from the command line?

### DISCUSSION THREAD

**Discussion by gracicot** [microsoft/vcpkg#39153] [Q&A] [answered]
I'm using a system where libstdc++ is only in a special path, and added to rpath by default by the compiler. Currenly, the `z_vcpkg_calculate_corrected_rpath` drops system paths so the required rpath to find libstdc++ is missing. I need to make it so that `X_VCPKG_RPATH_KEEP_SYSTEM_PATHS` is set to `ON` for all tool ports.

Is there a way to do that using the vcpkg command line or the environment? I'm I forced to create a port overlay for each tool port just to set that variable to `ON`?

**Comment by dg0yt** [comment, marked as answer]
You can set the variable to ON in a custom triplet.
You can use the custom triplet as host triplet if this is what you mean by "for all tool ports".

**Reply by gracicot** [reply]
Makes total sense and it works. I didn't know variables defined in triplets were forwarded to the scripts called in a portfile.

**Reply by dg0yt** [reply]
This may give a little more insight:
https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#useful-implementation-notes
